### PR TITLE
Np 1969 delete draft publication handler

### PR DESCRIPTION
**Pull request by janbj**

### CODE REVIEW

**Review by brinxmat — DISMISSED** [review]

File: delete-draft-publication-handler/src/test/java/no/unit/nva/publication/delete/DeleteDraftPublicationHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        when(environment.readEnv(DynamoDBPublicationService.TABLE_NAME_ENV))
+                .thenReturn(PublicationsDynamoDBLocal.NVA_RESOURCES_TABLE_NAME);
+        when(environment.readEnv(DynamoDBPublicationService.BY_PUBLISHER_INDEX_NAME_ENV))
+                .thenReturn(PublicationsDynamoDBLocal.BY_PUBLISHER_INDEX_NAME);
+        when(environment.readEnv(DynamoDBPublicationService.BY_PUBLISHED_PUBLICATIONS_INDEX_NAME))
+                .thenReturn(PublicationsDynamoDBLocal.BY_PUBLISHED_DATE_INDEX_NAME);
+    }
+
+    @Test
+    public void handleRequestDeletesPublicationWithStatusDraftForDeletion() throws ApiGatewayException {
```
Comment (axthosarouris) [inline delete-draft-publication-handler/src/test/java/no/unit/nva/publication/delete/DeleteDraftPublicationHandlerTest.java:75]: handleRequestDeletesPublicationWithoutDoiWhenStatusIsDraftForDeletion

File: delete-draft-publication-handler/src/test/java/no/unit/nva/publication/delete/DeleteDraftPublicationHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        handler.handleRequest(inputStream, outputStream, context);
+
+        NotFoundException exception = assertThrows(NotFoundException.class,
+            () -> publicationService.getPublication(publication.getIdentifier()));
+        String message = DynamoDBPublicationService.PUBLICATION_NOT_FOUND + publication.getIdentifier();
+        assertThat(exception.getMessage(), equalTo(message));
+    }
+
+    @Test
+    public void handleRequestDoesNotDeletePublicationWithStatusDraft() throws ApiGatewayException {
```
Comment (axthosarouris) [inline delete-draft-publication-handler/src/test/java/no/unit/nva/publication/delete/DeleteDraftPublicationHandlerTest.java:90]: unecessary exception

File: delete-draft-publication-handler/src/test/java/no/unit/nva/publication/delete/DeleteDraftPublicationHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+import static nva.commons.utils.IoUtils.inputStreamFromResources;
+import static nva.commons.utils.IoUtils.streamToString;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+@EnableRuleMigrationSupport
+public class DeleteDraftPublicationHandlerTest {
```
Comment (axthosarouris) [inline delete-draft-publication-handler/src/test/java/no/unit/nva/publication/delete/DeleteDraftPublicationHandlerTest.java:35]: What is the expected behavior when a publication has a draft doi and is drafted for deletion? Let's say there was a failure in deleting the draft doi

**Review by axthosarouris — CHANGES_REQUESTED** [review]

File: delete-draft-publication-handler/src/test/java/no/unit/nva/publication/delete/DeleteDraftPublicationHandlerTest.java (hunk 3 above)
Comment (janbj) [reply delete-draft-publication-handler/src/test/java/no/unit/nva/publication/delete/DeleteDraftPublicationHandlerTest.java:35]: This is a legal state until the deletion process has completed.

Comment (axthosarouris) [reply delete-draft-publication-handler/src/test/java/no/unit/nva/publication/delete/DeleteDraftPublicationHandlerTest.java:35]: But what is the handler's response?

File: template.yaml [hunk 4]
```diff
@@ -642,6 +642,39 @@ Resources:
...
+    DeleteDraftPublicationHandler:
+      Type: AWS::Serverless::Function
+      Properties:
+        Layers:
+          - !Ref NvaPublicationCommonsLayer
+        CodeUri: delete-draft-publication-handler
+        Handler: no.unit.nva.publication.delete.DeleteDraftPublicationHandler::handleRequest
+        Runtime: java11
+        MemorySize: 512
+        Role: !GetAtt LambdaRole.Arn
```
Comment (axthosarouris) [inline template.yaml:657]: You have to create a new role which allows the publication to delete items from the dynamo table.

**Review by brinxmat — APPROVED** [review]

**Review by axthosarouris — APPROVED** [review]

**Pull request merged**